Add Imviz documentation (#842)
spacetelescope/jdaviz: docs/imviz/displayimages.rst

.. _imviz-display-images:

*****************
Displaying Images
*****************

Imviz uses image viewers to visualize data from compatible files.

Selecting Data Set
==================

.. seealso::

>     `Selecting data set (Cubeviz) <https://jdaviz.readthedocs.io/en/latest/cubeviz/displaycubes.html#selecting-data-set>`_

Reviewer: Just one comment for now: Please use Sphinx `:ref:` for internal linking.

I'll review more tomorrow.

Author: Done.

.. _imviz-display-images:

*****************
Displaying Images
*****************

> Imviz uses image viewers to visualize data from compatible files.

Reviewer: ```suggestion
Imviz uses image viewers to visualize data from supported formats.
```

Selecting Data Set
==================

.. seealso::

    :ref:`Selecting Data Set<cubeviz-selecting-data>`

>         Documentation on selecting data sets in the jdaviz viewers.
> 

Reviewer: ```suggestion
        Documentation on selecting data sets in the Jdaviz viewers.
```

Adding New Viewers
==================

In the toolbar towards the top of the UI, there is a :guilabel:`picture with +` icon. By
clicking this icon, you can add new viewers to the application. You can then select from the data

> that has been loaded into the application to be visualized in these extra viewers. You can then utilize some of the

Reviewer: ```suggestion
that has been loaded into the application to be visualized in these additional viewers. You can then utilize some of the
```

Imviz-specific features, like Linked Pan/Zoom.

Defining Spatial Regions
========================

Spatial regions can be created the same way as in Cubeviz.

.. seealso::

>     :ref:`Defining Spatial Regions<spatial-regions>`

Reviewer: ```suggestion
    :ref:`Defining Spatial Regions <spatial-regions>`
```

        Documentation on defining spatial regions in an image viewer.

Blinking
========

Blinking is an Imviz-specific functionality that allows a user to quickly switch
between viewing two images. This can be done by selecting the :guilabel:`hammer and screwdriver` icon
and then clicking the :guilabel:`right arrow` icon. You can also access blinking

> by selecting the "B" key on your keyboard.

Reviewer: Uppercase might make user think they need to do Shift+b when this is not the case.
```suggestion
by selecting the "b" key on your keyboard.
```

Pan/Zoom and Linked Pan/Zoom
============================

Linked pan zoom is an Imviz-specific feature that allows the user to Pan and Zoom

> images in multiple different viewers simultaneously. This works by matching images
> based on their WCS (if present) or linking images by pixel.

Reviewer: This will be true after we merge #735.
```suggestion
images in multiple different viewers simultaneously. This works by matching images
based on the way they are linked together. Images are linked by pixels on load time,
but you can re-link them via WCS using `~jdaviz.configs.imviz.helper.Imviz.link_data`.
```

> Regular Pan/Zoom is also available and is used similarly to other jdaviz tools.

Reviewer: ```suggestion
Regular Pan/Zoom is also available and is used in a similar way as other Jdaviz tools.
```

.. seealso::

    :ref:`Pan/Zoom <cubeviz-pan-zoom>`

>         Documentation on using Pan/Zoom in the jdaviz viewers.

Reviewer: ```suggestion
        Documentation on using Pan/Zoom in the Jdaviz viewers.
```

.. _imviz-display-images:

*****************
Displaying Images
*****************

Imviz uses image viewers to visualize data from supported formats.

> Selecting Data Set

Reviewer 2: The Cubeviz section this links to is out of date, but I guess updating that can be a separate PR.

==================

.. seealso::

    :ref:`Selecting Data Set<cubeviz-selecting-data>`
        Documentation on selecting data sets in the Jdaviz viewers.

Adding New Viewers
==================

> In the toolbar towards the top of the UI, there is a :guilabel:`picture with +` icon. By

Reviewer 2: Any chance you can just include a small image of the icon here? "picture with +" is very unwieldy compared to just showing the icon.

clicking this icon, you can add new viewers to the application. You can then select from the data
that has been loaded into the application to be visualized in these additional viewers.
You can then utilize some of the Imviz-specific features, like Linked Pan/Zoom.

Defining Spatial Regions
========================

Spatial regions can be created the same way as in Cubeviz.

.. seealso::

    :ref:`Defining Spatial Regions <spatial-regions>`
        Documentation on defining spatial regions in an image viewer.

Blinking
========

Blinking is an Imviz-specific functionality that allows a user to quickly switch

> between viewing two images. This can be done by selecting the :guilabel:`hammer and screwdriver`

Reviewer 2: Is it two only, or two or more? Might be worth saying whether the images to blink between need to be selected in the data menu or not.

Author: @pllim ? I was not able to blink between more than two images, not sure if that is user error or not.

> icon and then clicking the :guilabel:`right arrow` icon. You can also access blinking

Reviewer 2: ```suggestion
icon and then clicking the :guilabel:`right arrow` icon. You can also blink
```

by selecting the "b" key on your keyboard.

Contrast/Bias
=============

In addition to changing contrast and bias information in the :guilabel:`layer` tab

> (see :ref:`Contrast <contrast>`), Imviz has a :guilabel:`white to black` button under the

Reviewer 2: Same comment as above, including the icon itself here might be nice if a small image can be included inline.

Author: Do you know how to add small images inline? I am only familiar with adding images to .rst files with `.. image::`.

:guilabel:`hammer and screwdriver` menu that can also edit those options. After clicking
that button, you can click and drag on the image viewer to change to change the contrast

> and bias. If you would like to reset to the default contrast and bias settings, you can

Reviewer 2: Which direction changes which property?

Author: I think that's a question for @pllim , I'd have to spend some time figuring it out.

Reviewer: https://github.com/spacetelescope/jdaviz/blob/79e4245663691a5f6567993be4c0198edebf4fbb/jdaviz/configs/imviz/plugins/tools.py#L150-L152

Drag across X changes bias, Y contrast. 😸

double-click.

Cursor Information
==================

By moving your cursor along the image viewer, you will be able to see information on the
cursor's location in pixel space (X and Y), the RA and Dec at that point, and the value

> there. This information is located in the top bar of the UI, on the

Reviewer 2: ```suggestion
of the data there. This information is located in the top bar of the UI, on the
```

.. _imviz-display-images:

*****************
Displaying Images
*****************

Imviz uses image viewers to visualize data from supported formats.

Selecting Data Set
==================

.. seealso::

    :ref:`Selecting Data Set<cubeviz-selecting-data>`
        Documentation on selecting data sets in the Jdaviz viewers.

Adding New Viewers
==================

> In the toolbar towards the top of the UI, there is a :guilabel:`picture with a plus` icon
> 
> .. image:: img/picture_with_plus.png
> 
> that when clicked will add new viewers to the application. You can then select from the data

Reviewer: I prefer the image to not break up the paragraph.

```suggestion
.. image:: img/picture_with_plus.png

In the toolbar towards the top of the UI, there is a :guilabel:`picture with a plus` icon
that when clicked will add new viewers to the application. You can then select from the data
```

that has been loaded into the application to be visualized in these additional viewers.

> You can then utilize some of the Imviz-specific features, like Linked Pan/Zoom.

Reviewer: ```suggestion
You can then utilize some of the Imviz-specific features, like :ref:`imviz_pan_zoom`.
```

Defining Spatial Regions
========================

Spatial regions can be created the same way as in Cubeviz.

.. seealso::

    :ref:`Defining Spatial Regions <spatial-regions>`
        Documentation on defining spatial regions in an image viewer.

Blinking
========

Blinking is an Imviz-specific functionality that allows a user to quickly switch

> between viewing two or more images, as long as they are linked. This can be done

Reviewer: I think this is good enough, unless you want a new section on Imviz linking?
```suggestion
between viewing two or more images, as long as they are linked
(see :ref:`imviz_pan_zoom` for more on linking behavior). This can be done
```

by selecting the :guilabel:`hammer and screwdriver` icon and then clicking the
:guilabel:`right arrow` icon. You can also blink by selecting the "b" key on your keyboard.

Contrast/Bias
=============

> In addition to changing contrast and bias information in the :guilabel:`layer` tab
> (see :ref:`Contrast <contrast>`), Imviz has a :guilabel:`white to black` button
> 
> .. image:: img/white_to_black.png
> 
> under the :guilabel:`hammer and screwdriver` menu that can also edit those options. After clicking

Reviewer: ```suggestion
.. image:: img/white_to_black.png

In addition to changing contrast and bias information in the :guilabel:`layer` tab
(see :ref:`Contrast <contrast>`), Imviz has a :guilabel:`white to black` button
under the :guilabel:`hammer and screwdriver` menu that can also edit those options. After clicking
```

that button, you can click and drag on the image viewer to change to change the contrast

> and bias. Moving along the x-axis will change the bias and moving along the y-axis will change the

Reviewer: You capitalized X and Y somewhere else, so might as well be consistent.
```suggestion
and bias. Moving along the X-axis will change the bias and moving along the Y-axis will change the
```

contrast. If you would like to reset to the default contrast and bias settings, you can

> double-click.

Reviewer: Can never be clear enough... 😆 

```suggestion
double-click on the display while the mode is active.
```

Cursor Information
==================

By moving your cursor along the image viewer, you will be able to see information on the
cursor's location in pixel space (X and Y), the RA and Dec at that point, and the value
of the data there. This information is located in the top bar of the UI, on the
middle-right side.

> 
> Pan/Zoom and Linked Pan/Zoom
> ============================

Reviewer: ```suggestion
.. _imviz_pan_zoom:

Pan/Zoom and Linked Pan/Zoom
============================
```
